File: client/src/api/banner.ts
import request from '@/utils/request';

// 배너 생성 (관리자)
export function createBanner(data: {
  title: string
  imageUrl: string
  linkUrl?: string
  displayOrder?: number
  status?: 'ACTIVE' | 'INACTIVE'
  startDate: string
  endDate: string
  regionSido?: string
  regionSigungu?: string
  targetPage?: 'HOME' | 'LEAGUE' | 'MATCH' | 'TEAM' | 'ALL'
}) {
  return request({
    url: '/banner',
    method: 'post',
    data,
  });
}

// 배너 수정 (관리자)
export function updateBanner(uid: string, data: {
  title?: string
  imageUrl?: string
  linkUrl?: string
  displayOrder?: number
  status?: 'ACTIVE' | 'INACTIVE'
  startDate?: string
  endDate?: string
  regionSido?: string
  regionSigungu?: string
  targetPage?: 'HOME' | 'LEAGUE' | 'MATCH' | 'TEAM' | 'ALL'
}) {
  return request({
    url: `/banner/${uid}`,
    method: 'put',
    data,
  });
}

// 배너 삭제 (관리자)
export function deleteBanner(uid: string) {
  return request({
    url: `/banner/${uid}`,
    method: 'delete',
  });
}

// 배너 상세 조회
export function getBannerDetail(uid: string) {
  return request({
    url: `/banner/${uid}`,
    method: 'get',
  });
}

// 배너 목록 조회 (관리자)
export function getBannerList(params?: {
  status?: 'ACTIVE' | 'INACTIVE'
  page?: number
  size?: number
}) {
  return request({
    url: '/banner',
    method: 'get',
    params,
  });
}

// 활성화된 배너 조회 (사용자)
export function getActiveBanners(params?: {
  targetPage?: 'HOME' | 'LEAGUE' | 'MATCH' | 'TEAM' | 'ALL'
  regionSigungu?: string
}) {
  return request({
    url: '/banner/active',
    method: 'get',
    params,
  });
}

File: backend/src/main/java/com/ttwijang/cms/entity/Banner.java
package com.ttwijang.cms.entity;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.Id;
import javax.persistence.PrePersist;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "banner")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Banner {

    @Id
    @Column(name = "uid", length = 50)
    private String uid;

    @Column(name = "title", length = 200, nullable = false)
    private String title;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(name = "link_url", length = 500)
    private String linkUrl;

    @Column(name = "display_order")
    private Integer displayOrder;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", length = 20, nullable = false)
    private BannerStatus status;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "region_sido", length = 50)
    private String regionSido;

    @Column(name = "region_sigungu", length = 50)
    private String regionSigungu;

    @Enumerated(EnumType.STRING)
    @Column(name = "target_page", length = 50)
    private TargetPage targetPage;

    @Column(name = "created_date")
    private LocalDateTime createdDate;

    @Column(name = "updated_date")
    private LocalDateTime updatedDate;

    @PrePersist
    public void prePersist() {
        if (this.uid == null) {
            this.uid = "banner-" + UUID.randomUUID().toString();
        }
        if (this.createdDate == null) {
            this.createdDate = LocalDateTime.now();
        }
        if (this.updatedDate == null) {
            this.updatedDate = LocalDateTime.now();
        }
        if (this.status == null) {
            this.status = BannerStatus.INACTIVE;
        }
        if (this.displayOrder == null) {
            this.displayOrder = 0;
        }
    }

    public enum BannerStatus {
        ACTIVE,
        INACTIVE
    }

    public enum TargetPage {
        HOME,
        LEAGUE,
        MATCH,
        TEAM,
        ALL
    }
}
